datadog/resource_datadog_synthetics_test_.go

```diff
@@ -466,24 +556,61 @@ func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *data
 	d.Set("request_headers", actualRequest.Headers)
 
 	actualAssertions := syntheticsTest.GetConfig().Assertions
-	var localAssertions []map[string]string
-	for _, assertion := range actualAssertions {
-		localAssertion := make(map[string]string)
-		if v, ok := assertion.GetOperatorOk(); ok {
-			localAssertion["operator"] = string(*v)
+	localAssertions := make([]map[string]interface{}, len(actualAssertions))
+	for i, assertion := range actualAssertions {
+		localAssertion := make(map[string]interface{})
+		if assertion.SyntheticsAssertionTarget != nil {
+			assertionTarget := assertion.SyntheticsAssertionTarget
+			if v, ok := assertionTarget.GetOperatorOk(); ok {
+				localAssertion["operator"] = string(*v)
+			}
+			if assertionTarget.HasProperty() {
+				localAssertion["property"] = assertionTarget.GetProperty()
+			}
+			if target := assertionTarget.GetTarget(); target != nil {
+				localAssertion["target"] = convertToString(target)
+			}
+			if v, ok := assertionTarget.GetTypeOk(); ok {
+				localAssertion["type"] = string(*v)
+			}
 		}
-		if assertion.HasProperty() {
-			localAssertion["property"] = assertion.GetProperty()
+		if assertion.SyntheticsAssertionJSONPathTarget != nil {
+			assertionTarget := assertion.SyntheticsAssertionJSONPathTarget
+			if v, ok := assertionTarget.GetOperatorOk(); ok {
+				localAssertion["operator"] = string(*v)
+			}
+			if assertionTarget.HasProperty() {
+				localAssertion["property"] = assertionTarget.GetProperty()
+			}
+			if target, ok := assertionTarget.GetTargetOk(); ok {
+				localTarget := make(map[string]string)
+				if v, ok := target.GetJsonPathOk(); ok {
+					localTarget["jsonpath"] = string(*v)
+				}
+				if v, ok := target.GetOperatorOk(); ok {
+					localTarget["operator"] = string(*v)
+				}
+				if v, ok := target.GetTargetValueOk(); ok {
+					localTarget["targetvalue"] = (*v).(string)
+				}
+				localAssertion["targetjsonpath"] = []map[string]string{localTarget}
+			}
+			if v, ok := assertionTarget.GetTypeOk(); ok {
+				localAssertion["type"] = string(*v)
+			}
 		}
-		if target := assertion.GetTarget(); target != nil {
-			localAssertion["target"] = convertToString(target)
+		localAssertions[i] = localAssertion
+	}
+	// If the config still uses assertions, keep using that in the state to not generate useless diffs
```

Comment:
I'd have to test around with this a bit more, but I'm not sure if `d` is always from the config. I think depending on which lifecycle piece we're in, `d` can reflect the object from the state instead.

Reply:
Yeah it mostly comes from the state, but that's what I care about here.